### PR TITLE
Feature/basic weather api data

### PR DESCRIPTION
**Pull request by LeaFranz**
Added first basic weather-card component with [weather-api](https://www.weatherapi.com) api on a view.
- api key should be in .env for now (see .env.example)
- added tanstack (same as https://github.com/sickinga/hux_weather/pull/2)
- added bootstrap vue for components

image

### CODE REVIEW

File: src/components/WeatherCard.vue [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+const fetchWeatherData = async (): Promise<AllWeatherData> => {
+    const url = `https://api.weatherapi.com/v1/forecast.json?q=${props.lat}%2C${props.lng}&days=3&lang=en&alerts=no&aqi=no&key=${process.env.VUE_APP_WEATHER_API}`
+
+    const response = await fetch(url)
+    if (!response.ok) {
+        // https://github.com/TanStack/query/issues/2258
+        throw new Error('Network response was not ok')
+    }
+    return response.json()
+}
```
Comment (sickinga) [inline src/components/WeatherCard.vue]: please put this in a separate hook file

File: src/types/weatherTypes.ts
Comment (sickinga) [inline src/types/weatherTypes.ts:1]: please name this `weather.d.ts`

File: src/utils/dateUtils.ts
Comment (sickinga) [inline src/utils/dateUtils.ts:1]: please name this ´date.utils.ts`

File: src/views/DashboardView.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    { value: "_c", text: 'Celsius' },
+    { value: '_f', text: 'Fahrenheit' },
+];
+
+</script>
+
+<template>
+    <div class="container">
+        <h1>Weather Dashboard</h1>
+        <b-form-select v-model="selectedUnit" :options="options" size="sm" class="unit-select"></b-form-select>
```
Comment (sickinga) [inline src/views/DashboardView.vue]: can this be a self closing tag?

**Review by sickinga — CHANGES_REQUESTED** [review]

File: src/components/WeatherCard.vue (hunk 1 above)
Comment (LeaFranz) [reply src/components/WeatherCard.vue]: should be added

**Review by sickinga — APPROVED** [review]

**Pull request merged**